quantity/core.py

```diff
@@ -228,9 +281,11 @@ def _operate(self, other, op, units_helper):
 
     # TODO: __dlpack__, __dlpack_device__
 
-    def __pow__(self, exp, mod=None):
-        exp = _check_pow_args(exp, mod)
-        if exp is NotImplemented:
+    def __pow__(self, exp: Any, mod: Any = None) -> Self | NotImplementedType:
+        if (mod := _parse_pow_mod(mod)) is NotImplemented:
```

Comment:
I like the symmetry of the parsing, but maybe we should just go for
```
if mod is not None:
    return NotImplemented
```
(and then perhaps overload `__pow__` itself??)